xarray/core/dataarray.py

```diff
@@ -3476,10 +3479,10 @@ def fillna(self, value: Any) -> Self:
 
     def interpolate_na(
         self,
-        dim: Hashable | None = None,
+        dim: Hashable,
```

Comment:
This seems like a breaking change, but I'm not sure what this does when passing None...

Reply:
Currently, it raises a `NotImplementedError`. Actually, this was never working, someone also raised an issue recently: #9392